src/main/java/io/cdap/plugin/successfactors/connector/SuccessFactorsConnector.java

```diff
@@ -0,0 +1,259 @@
+    throws IOException {
+    String entity = sampleRequest.getPath();
+    if (entity == null) {
+      throw new IllegalArgumentException("Path should contain entity name.");
+    }
+    try {
+      return listEntityData(entity, sampleRequest.getLimit());
+    } catch (EntityProviderException | SuccessFactorsServiceException | TransportException | EdmException e) {
+      throw new IOException("Unable to fetch data.", e);
+    }
+  }
+
+  /**
+   * @return returns the list of the data for the selected entity.
+   */
+  List<StructuredRecord> listEntityData(String entity, long top) throws SuccessFactorsServiceException,
+    IOException, TransportException, EntityProviderException, EdmException {
+    Edm edm = EntityProvider.readMetadata(getMetaDataStream(entity), false);
+    SuccessFactorsEntityProvider serviceHelper = new SuccessFactorsEntityProvider(edm);
+    EdmEntitySet edmEntitySet = serviceHelper.getEntitySet(entity);
+    InputStream dataStream = callEntityData(top, entity);
+    ODataFeed dataFeed = EntityProvider.readFeed(MediaType.APPLICATION_JSON, edmEntitySet,
+                                                 dataStream, EntityProviderReadProperties.init().build());
+    SuccessFactorsTransformer valueConverter = new SuccessFactorsTransformer(getSchema(entity));
+    List<ODataEntry> oDataEntryList;
+    oDataEntryList = dataFeed != null ? dataFeed.getEntries() : Collections.emptyList();
+    List<StructuredRecord> data = new ArrayList<>();
+    for (int i = 0; i < oDataEntryList.size(); i++) {
+      StructuredRecord dataRecord = valueConverter.buildCurrentRecord(oDataEntryList.get(i));
+      data.add(dataRecord);
+    }
+
+    return data;
+  }
+
+  /**
+   * @return returns the responseStream for the data of the selected entity.
+   */
+  private InputStream callEntityData(@Nullable long top, String entityName)
+    throws SuccessFactorsServiceException, TransportException, IOException, EntityProviderException, EdmException {
+    StringBuilder selectFields = new StringBuilder(String.join(PROPERTY_SEPARATOR, getNonNavigationalProperties
+      (entityName)));
+    URL dataURL = HttpUrl.parse(config.getBaseURL()).newBuilder().addPathSegment(entityName).
+      addQueryParameter(TOP_OPTION, String.valueOf(top)).addQueryParameter(SELECT_OPTION, selectFields.toString())
+      .build().url();
+    SuccessFactorsTransporter successFactorsHttpClient = new SuccessFactorsTransporter(config.getUsername(),
+                                                                                       config.getPassword());
+    SuccessFactorsResponseContainer responseContainer = successFactorsHttpClient.callSuccessFactorsWithRetry(dataURL);
+
+    ExceptionParser.checkAndThrowException("", responseContainer);
+    return responseContainer.getResponseStream();
+  }
+
+  List<String> getNonNavigationalProperties(String entity) throws TransportException, EdmException,
```

Comment:
where is this method called (other than tests)?

Reply:
private InputStream callEntityData(long top, String entityName) in this method.